### PR TITLE
Vellum demo in prelogin

### PR DESCRIPTION
**Pull request by orangejenny**
HQ support for https://github.com/dimagi/commcarehq-prelogin/pull/92

@benrudolph

### CODE REVIEW

File: corehq/apps/registration/views.py [hunk 1]
```diff
@@ -197,6 +200,13 @@ def register_user(request):
                     try:
                         requested_domain = request_new_domain(
                             request, form, is_new_user=True)
+                        if form.cleaned_data['xform']:
+                            lang = 'en'
+                            app = Application.new_app(requested_domain, "Untitled Application", application_version=APP_V2)
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/registration/views.py [hunk 2]
```diff
@@ -197,6 +200,14 @@ def register_user(request):
...
                         requested_domain = request_new_domain(
                             request, form, is_new_user=True)
+                        if form.cleaned_data['xform']:
+                            lang = 'en'
+                            app = Application.new_app(requested_domain, "Untitled Application",
+                                                      application_version=APP_V2)
+                            module = Module.new_module(_("Untitled Module"), lang)
+                            app.add_module(module)
+                            save_xform(app, app.new_form(0, "Untitled Form", lang), form.cleaned_data['xform'])
+                            app.save()
```
Comment (benrudolph) [inline corehq/apps/registration/views.py]: nice

**Comment by benrudolph** [comment]
this is awesome

File: corehq/apps/registration/views.py [hunk 3]
```diff
@@ -20,7 +20,12 @@
     track_workflow,
     track_confirmed_account_on_hubspot,
     track_clicked_signup_on_hubspot,
+    update_hubspot_properties
 )
+from corehq.apps.app_manager.const import APP_V2
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F401 'APP_V2' imported but unused

File: corehq/apps/registration/views.py [hunk 4]
```diff
@@ -20,7 +20,12 @@
...
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.apps.app_manager.models import Application, Module
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F401 'Application' imported but unused
F401 'Module' imported but unused

File: corehq/apps/registration/views.py [hunk 5]
```diff
@@ -20,7 +20,12 @@
...
+from corehq.apps.app_manager.util import save_xform
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F401 'save_xform' imported but unused

File: corehq/apps/registration/views.py [hunk 6]
```diff
@@ -91,6 +89,13 @@ def register_new_user(self, data):
                 request_new_domain(
                     self.request, reg_form, is_new_user=True
                 )
+                if form.cleaned_data['xform']:
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F821 undefined name 'form'

File: corehq/apps/registration/views.py [hunk 7]
```diff
@@ -91,6 +89,13 @@ def register_new_user(self, data):
...
+                    lang = 'en'
+                    app = Application.new_app(requested_domain, "Untitled Application", application_version=APP_V2)
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F821 undefined name 'requested_domain'

File: corehq/apps/registration/views.py [hunk 8]
```diff
@@ -91,6 +89,13 @@ def register_new_user(self, data):
...
+                    module = Module.new_module(_("Untitled Module"), lang)
+                    app.add_module(module)
+                    save_xform(app, app.new_form(0, "Untitled Form", lang), form.cleaned_data['xform'])
```
Comment (dimagimon) [inline corehq/apps/registration/views.py]: F821 undefined name 'form'

**Comment by nickpell** [comment]
merged with master; recommend checking that I did the merge correctly

**Comment by orangejenny** [comment]
@nickpell merge looks good, thanks for pinging

**Comment by orangejenny** [comment]
qa passed, synced with @amsagoff of analytics, this is good to go

File: corehq/apps/registration/views.py [hunk 9]
```diff
@@ -158,12 +174,16 @@ def ab(self):
 
     @property
     def page_context(self):
+        prefills = {
+            'email': self.prefilled_email,
```
Comment (benrudolph) [inline corehq/apps/registration/views.py]: one of these seems wrong, isn't it either `e` or `email`?

Comment (orangejenny) [inline corehq/apps/registration/views.py]: This is okay (just tested). Prelogin's email input is named `e` but HQ's signup form email input is `email`, which is where this is used.

Comment (benrudolph) [inline corehq/apps/registration/views.py]: gotcha thanks for checkign

**Pull request merged**